samples/meta-codegen/lib/src/test/java/com/my/company/codegen/MyclientcodegenGeneratorTest.java

```diff
@@ -0,0 +1,33 @@
+package com.my.company.codegen;
+
+import org.junit.Test;
+import org.openapitools.codegen.ClientOptInput;
+import org.openapitools.codegen.DefaultGenerator;
+import org.openapitools.codegen.config.CodegenConfigurator;
+
+/***
+ * This test allows you to easily launch your code generation software under a debugger.
+ * Then run this test under debug mode.  You will be able to step through your java code 
+ * and then see the results in the out directory. 
+ *
+ * To experiment with debugging your code generator:
+ * 1) Set a break point in MyclientcodegenGenerator.java in the postProcessOperationsWithModels() method.
+ * 2) To launch this test in Eclipse: right-click | Debug As | JUnit Test
+ *
+ */
+public class MyclientcodegenGeneratorTest {
+
+  @Test
+  public void launchCodeGenerator() {
+    // use this test to launch you code generator in the debugger.
+    // this allows you to easily set break points in MyclientcodegenGenerator.
```

Comment:
Maybe we should add a comment with a link to https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator-cli/src/main/java/org/openapitools/codegen/cmd/Generate.java so users can easily find sample logic?

Another option might be if we add a "helper" model in the main artifact for encapsulating options into an ADT and applying then to CodegenConfigurator, rather than having duplicate procedural logic between cli/maven/gradle. We could then use the same in these tests.

Reply:
I have added a comment.

I agree that some refactoring could be useful, but I do not have time to look at this right now.

Reply:
I'm working on a refactor of CodegenConfigurator similar to what I've suggested above.